synfig-studio/src/gui/app.cpp

```diff
@@ -2319,8 +2317,8 @@ App::quit()
 			return;
 	process_all_events();
 
-	Gtk::Main::quit();
-	// remove_window(*main_window);
+	// Gtk::Main::quit();
+	App::create()->remove_window(*main_window);
```

Comment:
You shouldn't do it: you're creating a new App here. Why?

AFAIK, the app survives as long as there is a ApplicationWindow open. When they are all closed (and we at Synfig will have only one), the Gtk::Application begins the shutdown, signalling on-shutdown, etc.

Reply:
Understood ! Kindly checkout the most recent change that I've made ! Its working ! no segmentation fault and fully functional ! :) Moving forward !